# Calculation of Enzyme and Vinyl Chloride
**Desired molar ratio of 1:6 which means 1 mol of VC to 6 mols of enzyme**

$$
\begin{align}
\text{Molar mass of vinyl chloride} &= 62.498 g/mole\\
\text{Mass (g)} &= N * M\\
N &= \frac m M \\
\end{align}
$$
Example with 40 $\mu g$ of VC
$$
\begin{align}
&= \frac{40 * 10^{-6}g}{62.498 g/mol}\\
&= 6.4002 * 10^{-7} mol
\end{align}
$$

Enzyme calculation
$$
\begin{align}
\text{Desired molar concentration of enzyme} &= 6 * 6.4002*10^{-6} mol = 3.84012*10^{-6} mol\\
\text{Molar mass of enzyme} &= 55000 g/mol\\
m &= N * M\\
&= 3.84012*10^{-6} mol * 55000 g/mol\\
&= 0.2112 g
\end{align}
$$


In [2]:
vc_volume = 20 # ul
vc_concentration = 2000 # ug / ml

vc_mass_formula = lambda x : (vc_concentration/1000)*x * 1e-6 #g
vc_mass = (vc_concentration/1000)*vc_volume * 1e-6 # g

vc_molar_mass = 62.498 # g/mole

vc_moles = vc_mass / vc_molar_mass # moles

desired_molar_concentration_enzyme = 6*vc_moles # moles

enzyme_molar_mass = 55000 # g/mole

enzyme_mass = desired_molar_concentration_enzyme * enzyme_molar_mass # g

print(
f'''
vc_mass = {vc_mass*1e6} micro g
vc_moles = {vc_moles*1e6:.5f} micro mol
enzyme_mass = {enzyme_mass:.4f} g
''')


vc_mass = 40.0 micro g
vc_moles = 0.64002 micro mol
enzyme_mass = 0.2112 g



## Recalculation of concentration
Since the calculation doesn't produce the same weight of enzyme required, but we know that the shown weight of enzyme ($0.272\mu g$) works, we are going to calculate backwards to get a new concentration instead. We are also assuming that the molar mass of the enzyme ($55000$ g/mole) is correct

Given that
$$
N = \frac m M
$$

In [3]:
known_enzyme_mass = 0.272 * 1e-6 # g

new_enzyme_mole = known_enzyme_mass / enzyme_molar_mass

display(new_enzyme_mole)

4.945454545454545e-12

This means that the concentration of the enzyme is 
$$
N = 4.945 \text{ pico mole} = 4.945 \cdot 10^{-12} mole
$$

Now we can get the concentration by dividing the enzyme moles with the VC moles to get

In [32]:
water_volume = 10 #ml
vc_concentration_desired = 5 # ug/ml
concentration_multiplier = 2

vc_weight = 40*1e-6 # g

concentration = (vc_moles / new_enzyme_mole)
vc_volume_2 = (vc_concentration_desired * water_volume)/(vc_concentration/1000) # ul

enzyme_weight = lambda x : ((x/vc_molar_mass)*concentration)*enzyme_molar_mass

def vc_weight_to_enzyme(vc_g):
    return ((vc_g * enzyme_molar_mass)/(concentration*vc_molar_mass))*concentration_multiplier

import pyperclip

output_string = f'''
| Property | Value |
| -------- | ----- |
|Enzyme mass|{enzyme_weight(vc_weight)} g|
|Concentration|{concentration}|
|Concentration multiplier|{concentration_multiplier}|
|Testing concentration|{(new_enzyme_mole * concentration) * vc_molar_mass }|

**Experiment properties**

| Property | Value |
| -------- | ----- |
| VC Volume | {vc_volume_2} ul |
| Enzyme required | {vc_weight_to_enzyme(vc_mass_formula(vc_volume_2))*1e6} micro grams |
|Water volume |{water_volume} ml |
| VC Concentration|{vc_concentration_desired} ug/ml|

'''
print(output_string)
print("--- String copied ---")
pyperclip.copy(output_string)


| Property | Value |
| -------- | ----- |
|Enzyme mass|4555.585670465049 g|
|Concentration|129415.90601487482|
|Concentration multiplier|2|
|Testing concentration|3.9999999999999996e-05|

**Experiment properties**

| Property | Value |
| -------- | ----- |
| VC Volume | 25.0 ul |
| Enzyme required | 0.6799999999999999 micro grams |
|Water volume |10 ml |
| VC Concentration|5 ug/ml|


--- String copied ---


#### Calculating equation of line
$$
y = mx + b\\
y-y_1 = (\frac{y_2 - y_1}{x_2 - x_1})(x-x_1)\\
y = (\frac{y_2 - y_1}{x_2 - x_1})(x-x_1) + y_1\\
$$

In [11]:
x = [0.1, 1]
y = [0.0004, 0.0031]

ri_predict = lambda a : ((y[1]-y[0])/(x[1]-x[0]))*(a-x[0])+y[0]
ri_predict(16)

0.04809999999999999